plugins/org.wso2.developerstudio.visualdatamapper/src/org/wso2/developerstudio/datamapper/servlets/RegistryReaderServlet.java

```diff
@@ -101,5 +109,30 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
             String result = IOUtils.toString(inputSchema);
             response.getWriter().println(result);
         }
+        
+    }
+
+    /**
+     * Saves schema changes to file in the datamapper editor
+     * 
+     */
+    private void saveChanges() {
+        Display.getDefault().syncExec(new Runnable() {
+            @Override
+            public void run() {
+                // Retrieves all unsaved editors
+                IEditorPart[] editors = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage()
+                        .getDirtyEditors();
+
+                for (int i = 0; i < editors.length; i++) {
+                    // Check if the editor is a datamapper editor
+                    if (editors[i].getEditorInput().getName().contains(DATAMAPPER_DIAGRAM_FILE_EXTENSION)) {
+                        editors[i].doSave(new NullProgressMonitor());
+                        break;
+                    }
+                }
+            }
+        });
     }
+        
 }
```

Comment:
As per WSO2 checkstyle standards, there should be an extra line at the end of the file. Again, I couldn't find any findbug plugin or checkstyle are enabled in the repo. Why it is not followed in this repo? 
@chanikag @dinuish94 ?

Reply:
Added new line in https://github.com/wso2/devstudio-tooling-esb/pull/1063

Reply:
There are several Eclipse frameworks, such as EMF, EEF, GMF, etc., that are being used in Integration Studio codebase. Each framework has its own code generations mechanism and we are developing on top of those generated code. However, some of the frameworks such as EEF, which is responsible for the properties view, does not comply with other formatters, instead, it generates following the default Eclipse formatter. Though we use WSO2 Eclipe formatter as the general practice, we cannot maintain a single formatting schema across the platform, as the code will be formatted again whenever the code is generated. It will be hard to maintain as we frequently generate the code for most of the changes. Therefore, we have disabled checkstyle at the moment. However, we are planning to enable findbugs with the current revamping efforts.

Reply:
OK, noted..